service/controller/resource/promxy/resource.go

```diff
@@ -0,0 +1,152 @@
+package promxy
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/pkg/relabel"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+type Config struct {
+	K8sClient    k8sclient.Interface
+	Logger       micrologger.Logger
+	Installation string
+	Provider     string
+}
+
+type Resource struct {
+	k8sClient    k8sclient.Interface
+	logger       micrologger.Logger
+	installation string
+	provider     string
+}
+
+func New(config Config) (*Resource, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Installation == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Installation must not be empty", config)
+	}
+
+	r := &Resource{
+		logger:       config.Logger,
+		k8sClient:    config.K8sClient,
+		installation: config.Installation,
+		provider:     config.Provider,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return "promxy"
+}
+
+func (r *Resource) toServerGroup(cluster metav1.Object) (*ServerGroup, error) {
+	httpClient := config.HTTPClientConfig{
+		TLSConfig: config.TLSConfig{
+			CAFile:             "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+			InsecureSkipVerify: true,
+		},
+		BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+	}
+
+	apiServerHost := r.k8sClient.RESTConfig().Host
+	apiServerURL, err := url.Parse(apiServerHost)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return &ServerGroup{
+		AntiAffinity:   time.Second * 10,
```

Comment:
What is this for? Maybe a comment would help.

Reply:
from docs: '# anti-affinity for merging values in timeseries between hosts in the server_group' - so, I'd actually dump this, we only have one server per servergroup, so we don't need to worry about anti-affinitiy